apps/sms/views/shared/js/utils.js

```diff
@@ -473,6 +473,25 @@
       });
       return parsed;
     },
+    url(base, params) {
+      if (base.indexOf('?') === -1) {
+        base += '?';
+      } else {
+        base += '&';
+      }
+
+      for (var key in params) {
+        if (params[key] == null) { // null or undefined
+          continue;
+        }
+
+        base +=
+          encodeURIComponent(key) + '=' +
+          encodeURIComponent(params[key]) + '&';
```

Comment:
I'll think a little more about it; I don't really like the sound of a URL that is not permanent; if the user bookmarks the page it wouldn't work later on.
Or maybe instead we should save a draft, and change the URL to be one that access the draft.